packages/ng-table-builder/src/components/table-thead/table-thead.component.ts

```diff
@@ -33,9 +33,18 @@ export class TableTheadComponent<T> {
 
     constructor(protected readonly filterable: FilterableService<T>) {}
 
+    public sortIfEnabled(): void {
+        const key: string | null | undefined = this.columnSchema?.key;
+        if (isNotNil(key) && (isTrue(this.columnSchema?.sortable) || isNotNil(this.sortableDefinition[key]))) {
```

Comment:
```ts
const sortIsEnabled: boolean = isNotNil(key) && (isTrue(this.columnSchema?.sortable) || isNotNil(this.sortableDefinition[key]));

if (sortIsEnabled) {
  this.sortByKey.emit(key);
}
```

Reply:
will not work because key possibly is `undefined`, i need to explicitly show that it is string;
 how about: 
```ts
    public sortIfEnabled(): void {
        const key: string | null | undefined = this.columnSchema?.key;
        if (isNotNil(key)) {
            const sortIsEnabled: boolean = isTrue(this.columnSchema?.sortable);
            const sortIsActive: boolean = isNotNil(this.sortableDefinition[key]);
            if (sortIsEnabled || sortIsActive) {
                this.sortByKey.emit(key);
            }
        }
    }
```